### DISCUSSION TITLE
More data through the official Streaming-API

### DISCUSSION THREAD

**Discussion by SebastianGode** [teslamate-org/teslamate#4407] [Ideas]
Tesla  has just announced their pricing for the Streaming-API. It includes a 10USD discount for each account which means you can use all API request up to 10USD every month without paying anything and they even have a spending-protection so that you don't pay anything in case you want to use more.

https://developer.tesla.com/de_DE

The 10USD a month are good for 500 data calls and 1.35 million Streaming-API signals.
If you calculate this you can do 25 Streaming signals every minute 24/7 for the whole month. With some logic that the streaming-requests are only sent when the car is charging or driving the resolution can be really high without paying anything. Maybe up to 75 signals per minute or 15 data fields per 5s.
As the new streaming-API includes data like battery temperature, current on DC charging, Autopilot status and co., I personally would love to switch over to it and log even more data than with the current old Owners API. 

How hard is it to set up all this stuff on my own and is Teslamate going to support new data fields now?

**Comment by Rathna-K** [comment]
to add one more question to this - 
2. Anyway to look at the metrics now to understand the current actual usage in our local env. And understand the estimated pricing?
